File: codes/java/chap05/answer_A32.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		// 入力
		int N = sc.nextInt();
		int A = sc.nextInt();
		int B = sc.nextInt();

		// 勝者を計算する
		boolean[] dp = new boolean[N + 1];
		for (int i = 0; i <= N; i++) {
			if (i >= A && dp[i-A] == false) dp[i] = true; // 勝ちの状態
			else if (i >= B && dp[i-B] == false) dp[i] = true; // 勝ちの状態
			else dp[i] = false; // 負けの状態
		}

		// 出力
		if (dp[N] == true) System.out.println("First");
		else System.out.println("Second");
	}
};

File: codes/python/chap05/answer_A32.py
# 入力
N, A, B = map(int, input().split())

# 配列 dp を定義
# dp[x]=True のとき勝ちの状態、dp[x]=False のとき負けの状態
dp = [ None ] * (N+1)

# 勝者を計算する
for i in range(N+1):
	if i >= A and dp[i-A] == False:
		dp[i] = True
	elif i >= B and dp[i-B] == False:
		dp[i] = True
	else:
		dp[i] = False

# 出力
if dp[N] == True:
	print("First")
else:
	print("Second")
